### PR TITLE
feat(turbopack): Handle fragment in requests

### PR DESCRIPTION
**Pull request by kdy1**
### Description

This reverts commit 1bfe729336b31bbaa74a3e38efd60c4faabaddd2.

### Testing Instructions

https://github.com/vercel/next.js/pull/64232

### CODE REVIEW

**Comment by kdy1** [comment]
I verified that the problematic test case, `test/e2e/app-dir/resource-url-encoding/resource-url-encoding.test.ts` , passes locally.

File: crates/turbopack-core/src/resolve/mod.rs [hunk 1]
```diff
@@ -1840,6 +1858,15 @@ async fn resolve_relative_request(
 
     let mut new_path = path_pattern.clone();
     if !options_value.fully_specified {
+        let fragment = fragment.await?;
+        if !fragment.is_empty() {
+            new_path.push(Pattern::Alternatives(
+                once(Pattern::Constant("".to_string()))
+                    .chain(once(Pattern::Constant(format!("#{fragment}"))))
```
Comment (sokra) [inline crates/turbopack-core/src/resolve/mod.rs:1865]: - this should also applied for fully specified requests
- in the case when we add the fragment to the path, we must no longer add it as fragment to the result

File: crates/turbopack-core/src/resolve/mod.rs [hunk 2]
```diff
@@ -1882,14 +1929,38 @@ async fn resolve_relative_request(
                                 options_value,
```
Comment (sokra) [inline crates/turbopack-core/src/resolve/mod.rs:1929]: You might need to handle fragment here too

**Review by sokra — COMMENTED** [review]
Looks good. Let's add a test case into `turbopack-tests/execution` too to verify its working

File: crates/turbopack-tests/tests/execution/turbopack/resolving/fragment/input/index.js [hunk 3]
```diff
@@ -0,0 +1,10 @@
+import client from "./client#component";
+import nofrag from "./nofrag#frag";
```
Comment (sokra) [inline crates/turbopack-tests/tests/execution/turbopack/resolving/fragment/input/index.js:2]: Maybe complete it with these two cases too

```
import client2 from "./client#component.js";
import nofrag2 from "./nofrag.js#frag";
```

**Review by sokra — APPROVED** [review]

**Pull request merged**